fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java

```diff
@@ -207,16 +222,20 @@ private String serializeSearchParmToQueryString(Parameter param) {
             while (process) {
                 param   = param.getNextParameter();
                 process = param != null && param.isChained();
+                // A guard is added as the original code 'could' result in an NPE.
                 if (param != null) {
-                    // A guard is added as the original code 'could' result in an NPE.
-                    returnString.append(param.getCode());
+                    if (param.isChained()) {
+                        returnString.append(SearchConstants.CHAINED_PARAMETER_CHARACTER).append(param.getCode());
```

Comment:
break this into two parts. 
returnString.append(SearchConstants.CHAINED_PARAMETER_CHARACTER) can go outside the if/else

and then you append(param.getCode) in the if
else you appendNormalParameter

Reply:
make sense, will change.

Reply:
done